crates/project/src/project.rs

```diff
@@ -6612,10 +6612,30 @@ impl Project {
             let file = File::from_dyn(buffer.file()).and_then(File::as_local);
             if let (Some(file), Some(language_server)) = (file, language_server) {
                 let lsp_params = request.to_lsp(&file.abs_path(cx), buffer, &language_server, cx);
+                let status = request.status();
                 return cx.spawn(move |this, cx| async move {
                     if !request.check_capabilities(language_server.capabilities()) {
                         return Ok(Default::default());
                     }
+                    let id = language_server.next_id().load(SeqCst);
```

Comment:
I think this approach of grabbing the next ID seems error prone. An alternative design could be that we create a custom struct that implements `Future`, that gets returned from `language_server.request()`, that also stores the ID on itself. That way we can re-arrange this logic such that we call `language_server.request()`, get the ID off of it, and then await it after we've put in the language server progress.